### PR TITLE
fix a panic on missing DigitalOcean credentials

### PR DESCRIPTION
**Pull request by kdomanski**
```release-note
NONE
```

### CODE REVIEW

File: api/pkg/validation/cluster.go [hunk 1]
```diff
@@ -386,6 +386,10 @@ func validateAzureCloudSpec(spec *kubermaticv1.AzureCloudSpec) error {
 
 func validateDigitaloceanCloudSpec(spec *kubermaticv1.DigitaloceanCloudSpec) error {
 	if spec.Token == "" {
```
Comment (alvaroaleman) [inline api/pkg/validation/cluster.go:388]: Wouldn't a `if spec.Token == "" && spec.CredentialsReference == nil {` make more sense?

Comment (kdomanski) [reply api/pkg/validation/cluster.go:388]: It's perfectly equivalent.

**Review by alvaroaleman — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**